datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxExternalDatabaseConfigurer.java

```diff
@@ -50,4 +59,24 @@ private void setExperimentalForCreateDatabaseIfNeeded(SdxDatabaseRequest sdxData
             sdxCluster.setCreateDatabase(true);
         }
     }
+
+    private boolean isCMExternalDbSupported(CloudPlatform cloudPlatform, SdxCluster sdxCluster) {
+        if (CloudPlatform.AZURE == cloudPlatform) {
+            String runtime = sdxCluster.getRuntime();
+            if (StringUtils.isBlank(runtime)) {
+                LOGGER.info("Runtime is not specified, external DB is permitted on Azure");
+                return true;
+            }
+            boolean permitted = isVersionNewerOrEqualThan(sdxCluster::getRuntime, () -> AZURE_EXT_DB_MIN_RUNTIME_VERSION);
+            LOGGER.info("External DB {} permitterd on Azure with runtime version: {}", permitted ? "is" : "is NOT", runtime);
+            return permitted;
+        }
+        return true;
+    }
+
+    private boolean isVersionNewerOrEqualThan(Versioned currentVersion, Versioned baseVersion) {
+        LOGGER.info("Compared: version {} with new version {}", currentVersion.getVersion(), baseVersion.getVersion());
+        Comparator<Versioned> versionComparator = new VersionComparator();
+        return versionComparator.compare(currentVersion, baseVersion) > -1;
```

Comment:
Since `VersionComparator` is stateless, I suggest instantiating it only once as a bean or field initializer in this class, instead of creating a new instance each time this method is invoked.